### PR TITLE
feat: new lint rule to enforce stylex file extension

### PR DESCRIPTION
**Pull request by mellyeliu**
## Context
This ESLint rule ensures consistency in file naming conventions when exporting `stylex.defineVars`. 

Specifically:
- Files that export `stylex.defineVars` must end with `.stylex.jsx` or `.stylex.tsx` 
- Files that do not export `stylex.defineVars` must not use the `.stylex.jsx` or `.stylex.tsx` extensions
- Users can configure the file extension via a themeFileExtension option (e.g., `.custom.jsx`), defaulting to `.stylex.jsx`.
## Test Cases:
Valid cases:
- Files with `stylex.defineVars` export correctly using `.stylex.jsx`, `.stylex.tsx`, or custom extensions.
- Files without `stylex.defineVars` export using non-.stylex extensions.

Invalid cases:
- Files exporting `stylex.defineVars` without the correct extension.
- Files not exporting `stylex.defineVars` but using `.stylex.jsx/.tsx` or custom extensions incorrectly.

- [x] I have read the contributing guidelines
      [Contribution Guidelines](https://github.com/facebook/stylex/blob/main/.github/CONTRIBUTING.md)
- [x] Performed a self-review of my code

### CODE REVIEW

File: packages/eslint-plugin/__tests__/stylex-enforce-extension-test.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        },
+      ],
+    },
+    {
+      code: 'export const somethingElse = {};',
+      filename: 'testComponent.stylex.tsx',
+      errors: [
+        {
+          message:
+            'Files that do not export `stylex.defineVars` must not have a `.stylex.jsx` or `.stylex.tsx` extension.',
```
Comment (nmn) [inline packages/eslint-plugin/__tests__/stylex-enforce-extension-test.js:91]: Change the error message please:

```
Only StyleX variables defined with `stylex.defineVars()` can be exported from a file with the ".stylex.tsx" extension.
```

File: packages/eslint-plugin/src/stylex-enforce-extension.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+            if (
+              init &&
+              init.callee &&
+              init.callee.object &&
+              init.callee.object.name === 'stylex' &&
+              init.callee.property &&
+              init.callee.property.name === 'defineVars'
+            ) {
+              hasDefineVarsExport = true;
```
Comment (nmn) [inline packages/eslint-plugin/src/stylex-enforce-extension.js:65]: Not quite. This just verifies that any `.stylex.js` file has at least one `stylex.defineVars`. We want to make sure there *no other exports*.

You shouldn't need the `Program:exit` at all. You should only need `ExportNamedDeclrataion` and `ExportDefaultDeclaration`.

**Review by nmn — CHANGES_REQUESTED** [review]
Not quite what we want.

File: packages/eslint-plugin/src/stylex-enforce-extension.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        if (hasDefineVarsExport && !isStylexFile) {
+          context.report({
+            node,
+            message: `Files that export \`stylex.defineVars\` must have a \`${themeFileExtension}\` or \`${themeTsxExtension}\` extension.`,
+          });
+        } else if (!hasDefineVarsExport && isStylexFile) {
+          context.report({
+            node,
+            message: `Files that do not export \`stylex.defineVars\` must not have a \`${themeFileExtension}\` or \`${themeTsxExtension}\` extension.`,
+          });
```
Comment (nmn) [inline packages/eslint-plugin/src/stylex-enforce-extension.js:84]: We also want the errors to be reported on the export statements and not on the file level.

File: packages/eslint-plugin/src/stylex-enforce-extension.js (hunk 2 above)
Comment (mellyeliu) [reply packages/eslint-plugin/src/stylex-enforce-extension.js]: thanks for the requirements clarification here - updated to explicitly forbid mixed imports

File: packages/eslint-plugin/src/stylex-enforce-extension.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+              init.callee.object.name === 'stylex' &&
+              init.callee.property &&
+              init.callee.property.name === 'defineVars'
+            ) {
+              hasStyleXVarsExports = true;
+            } else {
+              hasOtherExports = true;
+            }
+          });
+        }
```
Comment (nmn) [inline packages/eslint-plugin/src/stylex-enforce-extension.js:65]: You can do something simple like this.

```suggestion
        if (node.declaration && node.declaration.declarations) {
          node.declaration.declarations.forEach((declaration) => {
            const init = declaration.init;

            if (
              init &&
              init.callee &&
              init.callee.object &&
              init.callee.object.name === 'stylex' &&
              init.callee.property &&
              init.callee.property.name === 'defineVars' &&
              !isStyleXFile
            ) {
              context.report({
                node: declaration.init,
                message: "...Must be in a .stylex.js file"
              });
            } else if (isStyleXFile) {
              context.report({
                node: declaration.init,
                message: "...Must NOT be in a .stylex.js file"
              });
            }
          });
        }
```

**Review by nmn — CHANGES_REQUESTED** [review]
You missed Default Exports. Default exports should be treated all other exports.

**Comment by mellyeliu** [comment]
Processing for default exports, general refactor to shared functions

**Review by nmn — APPROVED** [review]
Nice big improvement. Thanks!

**Pull request merged**